device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -1036,10 +1045,12 @@ private void checkForNewlyUnregisteredMultiplexedClientsToStop()
             }
 
             configsUnregisteredSuccessfully.add(configToUnregister);
+            this.deviceSessionsOpenedLatches.remove(configToUnregister.getDeviceId());
             configToUnregister = configsToUnregisterIterator.hasNext() ? configsToUnregisterIterator.next() : null;
         }
 
         this.multiplexingClientsToUnregister.removeAll(configsUnregisteredSuccessfully);
+        this.deviceClientConfigs.removeAll(configsUnregisteredSuccessfully);
```

Comment:
Same as above. Some local state that wasn't made up to date when a device is unregistered